test/test_macho.rb

```diff
@@ -46,6 +46,15 @@ def test_load_commands
     end
   end
 
+  def test_load_command_string_malformed
+    filename = fixture(:i386, "hello.bin")
+    file = MachO::MachOFile.new(filename)
+
+    assert_raises MachO::LCStrMalformedError do
+      file.change_install_name(file.linked_dylibs.first, "\x00bad")
```

Comment:
I was confused for a moment why this indeed raises the exception, but it does so after writing the malformed `LCStr` and when re-reading all load commands. (I think we should be preventing this problematic user input earlier, but as stated in my other comment, not as part of this PR.)